src/integrations/metamask/index.js

```diff
@@ -50,6 +50,10 @@ class Metamask extends InjectedWeb3 {
       this.walletEnabled = false
     }
 
+    if (this.watcher) {
```

Comment:
Code LGTM but I suggest to refactor the the ```this.walletEnabled``` stuff. This file is 80 LOC and it would not be a problem if you just create a function for that. Something like:
```
this.walletEnabled = initWallet()
....
const initWallet = () => {
  try {
    if (!...) {
        return false
    }
    // init provider
} catch (err) {
    return false
}
```